config_api.go

```diff
@@ -17,9 +20,44 @@ type ClientBinary struct {
 }
 
 type ClientConfig struct {
-	ClientBinaries   []ClientBinary `yaml:"client_binaries" json:"client_binaries"`
-	S3ClientBinaries []ClientBinary `yaml:"s3_client_binaries" json:"s3_client_binaries"`
-	LatestRevision   string         `yaml:"latest_revision" json:"latest_revision"`
+	ClientBinaries   []ClientBinary
+	S3ClientBinaries []ClientBinary
+	LatestRevision   string
+	S3URLPrefix      string
+}
+
+func (c *ClientConfig) populateClientBinaries(ctx context.Context, bucket pail.Bucket, prefix string) error {
+	if !strings.HasSuffix(prefix, "/") {
+		prefix += "/"
+	}
+
+	iter, err := bucket.List(ctx, prefix)
+	if err != nil {
+		return errors.Wrap(err, "listing client bucket")
+	}
+	for iter.Next(ctx) {
+		item := strings.TrimPrefix(iter.Item().Name(), prefix)
+		name := strings.Split(item, "/")
```

Comment:
It seems like all of this string splitting relies on the bucket names being a certain format -- just for reference, could we put an example or the template of what we're expecting in a comment? (Seems like <prefix/<display_name_that's_of_some_specific_format>/<os>_<arch>/evergreen if i'm parsing that right)

Reply:
added